src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -30194,6 +30198,105 @@ public static int TakeOutParam<T>(out T x)
             Assert.True(x1.Type.IsErrorType());
         }
 
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/17377")]
+        public void GlobalCode_InferenceFailure_07()
+        {
+            string source =
+@"
+H.M((var x1, int x2));
+H.M(x1);
+
+class H
+{
+    void M(object a) {}
+}
+";
+            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.ReleaseExe.WithScriptClassName("Script"), parseOptions: TestOptions.Script);
+
+            compilation.GetDeclarationDiagnostics().Verify(
```

Comment:
I am adding code to assert all errors. However, note that this test is for a scenario that is not intended to be fixed in this PR, and remains skipped.